### PR TITLE
Update linters

### PR DESCRIPTION
**Pull request by drewdrewthis**

### CODE REVIEW

**Comment by jmcolella** [comment]
@drewdrewthis @cisacke @patrickkim 👀  please. Markdown linter is not finished, but want to push through CSS/JS linters

**Comment by jmcolella** [comment]
@drewdrewthis @cisacke @patrickkim OK, this changes a lot. Main take aways:

1. eslint configured using airbnb configuration + custom rules
2. stylelint configured for linting CSS
3. esling markdown configured for Styleguidist
4. linter run for every commit/push (will not proceed if warnings/failures)
5. linter errors WILL NOT stop development build

File: .eslintrc [hunk 1]
```diff
@@ -1,79 +1,155 @@
...
+    'arrow-spacing': [ 1, {
+      'before': true,
+      'after': true
+    }],
+    'arrow-body-style': [
+      2,
+      'as-needed'
+    ],
     'block-scoped-var': 1,
+    'block-spacing': 1,
```
Comment (j-clark) [inline .eslintrc]: IME, warnings will be ignored and will just be noise. can we make them errors / fix violations or just remove the rule?

File: package.json [hunk 2]
```diff
@@ -66,9 +82,18 @@
     "react-dom": "^15.5.4",
     "react-hot-loader": "^1.3.1",
     "react-styleguidist": "^5.0.6",
+    "redux-saga": "^0.14.6",
```
Comment (j-clark) [inline package.json]: we shouldn't need redux-saga in this project

File: .eslintrc [hunk 3]
```diff
@@ -1,79 +1,155 @@
...
+    'brace-style': 2,
+    'camelcase': 2,
+    'class-methods-use-this': 0,
+    'comma-dangle': [
+      0,
+      'always-multiline'
+    ],
+    'comma-style': 1,
+    'computed-property-spacing': 1,
     'curly': [ 2, 'all' ],
```
Comment (patrickkim) [inline .eslintrc]: Can we lax this rule to `'curly': ["error", "multi-line"],`?

Comment (patrickkim) [inline .eslintrc]: I would like to be able to support this

```js
function setWeight ( { semibold, light } ) {
  if ( semibold ) return styles.semibold;
  if ( light ) return styles.light;

  return styles.regular;
}
```

**Pull request merged**